app/knight.py

```diff
@@ -0,0 +1,53 @@
+class Knight:
+    def __init__(self, config: dict) -> None:
+        self.name = config["name"]
+        self.power = self.apply_power(config)
+        self.protection = self.apply_armour(config)
+        self.hp = self.apply_hp(config)
```

Comment:
The methods `apply_power`, `apply_armour`, and `apply_hp` assume that certain keys exist in the config dictionary. To prevent potential KeyErrors, consider using `config.get('key', default_value)` to provide default values if keys are missing.